model_compression_toolkit/data_generation/common/data_generation.py

```diff
@@ -85,19 +87,19 @@ def data_generation(
             input_imgs = image_pipeline.image_input_manipulation(imgs_to_optimize)
 
             # Forward pass to extract activations
-            output = activation_extractor.run_on_inputs(input_imgs)
+            output = activation_extractor.run_model(input_imgs)
 
             # Compute BatchNorm alignment loss
             bn_loss = all_imgs_opt_handler.compute_bn_loss(input_imgs=input_imgs,
                                                            batch_index=random_batch_index,
                                                            activation_extractor=activation_extractor,
                                                            orig_bn_stats_holder=orig_bn_stats_holder,
                                                            bn_alignment_loss_fn=bn_alignment_loss_fn,
-                                                           layer_weights=layer_weights)
+                                                           bn_layer_weights=bn_layer_weights)
+
 
             # Compute output loss
-            output_loss = output_loss_fn(output_imgs=output,
-                                         output_loss_multiplier=output_loss_multiplier)
+            output_loss = output_loss_fn(output_imgs=output) if output_loss_multiplier > 0 else torch.zeros(1).to(get_working_device())
```

Comment:
put the "if" above the output_loss_fn. No need to generate the zeros vector

Reply:
I add it to total loss so there is a need for this